src/Http/Ownership/RequestOwnershipRequestHandler.php

```diff
@@ -93,12 +83,31 @@ public function handle(ServerRequestInterface $request): ResponseInterface
             }
         }
 
+        // Make sure the current user is allowed to request ownership for this organizer
         $this->ownershipStatusGuard->isAllowedToRequest(
             $requestOwnership->getItemId()->toString(),
             $requestOwnership->getOwnerId()->toString(),
             $this->currentUser
         );
 
+        // Make sure there is no open request for this item and owner
+        $existingOwnershipItems = $this->ownershipSearchRepository->search(
+            new SearchQuery([
+                new SearchParameter('itemId', $requestOwnership->getItemId()->toString()),
+                new SearchParameter('ownerId', $requestOwnership->getOwnerId()->toString()),
+            ])
+        );
+
+        /** @var OwnershipItem $ownershipItem */
+        foreach ($existingOwnershipItems as $ownershipItem) {
+            if ($ownershipItem->getState() === OwnershipState::requested()->toString() ||
+                $ownershipItem->getState() === OwnershipState::approved()->toString()) {
+                throw ApiProblem::ownerShipAlreadyExists(
+                    'An ownership request for this item and owner already exists with id ' . $ownershipItem->getId()
```

Comment:
I see no unit test for this case, there is a feature test tho

Reply:
If you check `RequestOwnershipRequestHandlerTest` there are in total 4 tests that take into account the state. 

There are 2 tests that work because the state is either rejected or deleted:
- it_allows_requesting_same_ownership_when_rejected
- it_allows_requesting_same_ownership_when_deleted

There are 2 tests that test the requested and approved states:
- it_prevents_requesting_same_ownership_when_approved
- it_prevents_requesting_same_ownership (maybe less explicit about the state)